test/SlowTests/Issues/RavenDB-18554.cs

```diff
@@ -0,0 +1,244 @@
+using Raven.Client.Documents.Session;
+using Raven.Client.Exceptions.Database;
+using Raven.Client.ServerWide;
+using Raven.Client.ServerWide.Operations;
+using Raven.Server;
+using Raven.Server.Commercial;
+using Raven.Server.Documents;
+using Raven.Server.Documents.Indexes;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_18554 : ClusterTestBase
+    {
+        public RavenDB_18554(ITestOutputHelper output) : base(output)
+        {
+        }
+
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task QueriesShouldFailoverIfDatabaseIsCompacting(bool cluster = false)
+        {
+            Options storeOptions;
+            RavenServer leader = null;
+            List<RavenServer> nodes = null;
+            if (cluster)
+            {
+                (nodes, leader) = await CreateRaftCluster(2);
+                storeOptions = new Options {Server = leader, ReplicationFactor = nodes.Count, RunInMemory = false};
+            }
+            else
+            {
+                storeOptions = new Options {RunInMemory = false};
+            }
+            
+            using (var store = GetDocumentStore(storeOptions))
+            {
+                // Prepare Server For Test
+                string categoryId;
+                Category c = new Category { Name = $"n0", Description = $"d0" };
+                using (var session = store.OpenAsyncSession())
+                {
+                    await session.StoreAsync(c);
+                    if(cluster)
+                        session.Advanced.WaitForReplicationAfterSaveChanges(replicas: 1);
+                    await session.SaveChangesAsync();
+                    categoryId = c.Id;
+                }
+                
+                var index = new Categoroies_Details();
+                index.Execute(store);
+                Indexes.WaitForIndexing(store);
+
+                // Test
+                CompactSettings settings = new CompactSettings {DatabaseName = store.Database, Documents = true, Indexes = new[]{ index.IndexName } };
+
+                Exception exception = null;
+                List<Categoroies_Details.Entity> l = null;
+                var d = () =>
+                {
+                    try
+                    {
+                        using (var session = store.OpenSession())
+                        {
+                            l = session.Query<Categoroies_Details.Entity, Categoroies_Details>()
+                                .ProjectInto<Categoroies_Details.Entity>()
+                                .ToList();
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        exception = e;
+                    }
+                };
+
+                if (cluster)
+                {
+                    foreach (var node in nodes)
```

Comment:
You shouldn't set the task for all nodes in the cluster.
Only the relevant node.